src/openrct2-ui/windows/ViewClipping.cpp

```diff
@@ -303,8 +303,8 @@ static void window_view_clipping_update(rct_window* w)
     if (_toolActive && !window_view_clipping_tool_is_active())
     {
         _toolActive = false;
-        gClipSelectionA = _previousClipSelectionA;
-        gClipSelectionB = _previousClipSelectionB;
+        gClipSelectionA = { static_cast<uint8_t>(_previousClipSelectionA.x), static_cast<uint8_t>(_previousClipSelectionA.y) };
```

Comment:
Why are all these casts necessary?

Reply:
`gClipSelectionA` is a `LocationXY8` struct, so I'm making the narrowing explicit.

Reply:
Looks like gClipSelectionA / B contains tile coordinates - can you modify it to contain TileCoords?

Reply:
I modified `_previousClipSelectionA` and `_previousClipSelectionB` to be `TileCoordsXY`. I didn't touch `gClipSelection(A|B)` because those are bigger changes that even involve some comparisons to sprite coords and I had let them out of this PR on purpose.